// @pixi/ui/lib/Layout.mjs
import { Container } from '@pixi/display';

class Layout extends Container {
  constructor(options) {
    super();
    /** TODO */
    this.children = [];
    this.options = options;
    if (options?.type) {
      this.type = options.type;
    }
    if (options?.children) {
      options.children.map((child) => this.addChild(child));
    }
  }
  onChildrenChange() {
    let x = this.options?.horPadding ?? 0;
    let y = this.options?.vertPadding ?? 0;
    const elementsMargin = this.options?.elementsMargin ?? 0;
    this.children.forEach((child) => {
      if (!this.type && x + child.width >= this.parent.width) {
        y += elementsMargin + child.height;
        x = this.options?.horPadding ?? 0;
        child.x = x;
        child.y = y;
        this.h = y;
      } else {
        child.x = x;
        child.y = y;
      }
      switch (this.type) {
        case "horizontal":
          x += elementsMargin + child.width;
          this.w = x;
          this.h = child.height;
          break;
        case "vertical":
          y += elementsMargin + child.height;
          this.h = y;
          this.w = child.width;
          break;
        default:
          x += elementsMargin + child.width;
          this.w = x;
          break;
      }
    });
  }
}
function row(elementsMargin, ...params) {
  const children = [];
  params.forEach((param) => children.push(param));
  return new Layout({
    type: "horizontal",
    elementsMargin,
    children
  });
}
function col(elementsMargin, ...params) {
  const children = [];
  params.forEach((param) => children.push(param));
  return new Layout({
    type: "vertical",
    elementsMargin,
    children
  });
}

export { Layout, col, row };
